tests/distro/repository_validator/application/test_dpkg_url_validator.py

```diff
@@ -0,0 +1,130 @@
+def test_validator_when_repo_does_not_exists():
+    """
+    Test case for the validator when the repository does not exist.
+
+    This test case checks the behavior of the validator when the repository does not exist.
+    It validates that the validator raises a PackageDoesNotExist exception.
+
+    Steps:
+    1. Create an instance of InMemoryPackageRepository with an empty repository URL list.
+    2. Create an instance of DPKGUrlValidator with the repository.
+    3. Invoke the validator with the required parameters.
+    4. Check that the validator raises a PackageDoesNotExist exception.
+    """
+    with pytest.raises(PackageDoesNotExist) as package_error:
+        repository = InMemoryPackageRepository([])
+        validator = DPKGUrlValidator(repository=repository)
+
+        validator(
+            name="ednx_package",
+            version="2.0.1",
+            domain="github.com",
+            extra={"path": "test",
+                    "protocol": "https",
+                    "repo": "testrepo"}
+        )
+
+    # Then
+    assert package_error.type is PackageDoesNotExist
+
+
+# Test case for validator when extra pip requirements exist
+def test_validator_when_extra_pip_requeriments_exists():
+    """
+    Test case for the validator when extra pip requirements exist.
+
+    This test case checks the behavior of the validator when extra pip requirements exist.
+    It validates that the validator does not raise any exceptions.
+
+    Steps:
+    1. Create an instance of InMemoryPackageRepository with a non-empty repository URL list.
+    2. Create an instance of ExtraPipRequirementsUrlValidator with the repository.
+    3. Invoke the validator with a valid URL.
+    """
+    repository = InMemoryPackageRepository(
+        ["https://github.com/eduNEXT/edx_xblock_scorm/tree/v2.0.0"]
+    )
+    validator = ExtraPipRequirementsUrlValidator(repository=repository)
+    validator(
+        url="git+https://github.com/eduNEXT/edx_xblock_scorm@v2.0.0#egg=scormxblock-xblock==2.0.0"
+    )
+
+
+def test_validator_when_extra_pip_requeriments_does_not_exists():
+    """
+    Test case for the validator when extra pip requirements do not exist.
+
+    This test case checks the behavior of the validator when extra pip requirements do not exist.
+    It validates that the validator raises a PackageDoesNotExist exception.
+
+    Steps:
+    1. Create an instance of InMemoryPackageRepository with an empty repository URL list.
+    2. Create an instance of ExtraPipRequirementsUrlValidator with the repository.
+    3. Invoke the validator with an invalid URL.
+    4. Check that the validator raises a PackageDoesNotExist exception.
+    """
+    with pytest.raises(PackageDoesNotExist) as package_error:
+        repository = InMemoryPackageRepository([])
+        validator = ExtraPipRequirementsUrlValidator(repository=repository)
+
+        validator(
+            url="git+https://github.com/eduNEXT/edx_xblock_scorm@v2.0.0"
```

Comment:
could we use a fake url instead of a real url for test?

Reply:
Done